controller/array_action/array_mediator_svc.py

```diff
@@ -254,8 +256,40 @@ def get_volume(self, volume_name, pool_id=None):
     def get_volume_name(self, volume_id):
         return self._get_volume_name_by_wwn(volume_id)
 
+    def _get_object_fcmaps(self, object_name):
+        all_fcmaps = []
+        fcmap_as_target = self._get_fcmap_as_target_if_exists(object_name)
+        if fcmap_as_target:
+            all_fcmaps.append(fcmap_as_target)
+        all_fcmaps.extend(self._get_fcmaps_as_source_if_exist(object_name))
+        return all_fcmaps
+
+    def _expand_cli_volume(self, cli_volume, expanded_bytes):
+        volume_name = cli_volume.name
+        try:
+            self.client.svctask.expandvdisksize(vdisk_id=volume_name, unit='b', size=expanded_bytes)
+        except (svc_errors.CommandExecutionError, CLIFailureError) as ex:
+            if not is_warning_message(ex.my_message):
+                logger.warning("Failed to expand volume {}".format(volume_name))
+                if OBJ_NOT_FOUND in ex.my_message or VOL_NOT_FOUND in ex.my_message:
+                    raise controller_errors.ObjectNotFoundError(volume_name)
+                if NOT_ENOUGH_EXTENTS_IN_POOL_EXPAND in ex.my_message:
+                    raise controller_errors.NotEnoughSpaceInPool(pool=cli_volume.mdisk_grp_name)
+                else:
+                    raise ex
+        except Exception as ex:
+            logger.exception(ex)
+            raise ex
+
     def expand_volume(self, volume_id, required_bytes):
-        pass
+        cli_volume = self._get_cli_volume_by_wwn(volume_id, not_exist_err=True)
+        volume_name = cli_volume.name
+        fcmaps = self._get_object_fcmaps(volume_name)
+        self._safe_delete_fcmaps(volume_name, fcmaps)
+        volume_current_size = int(cli_volume.capacity)
+        final_required_size = self._convert_size_bytes(required_bytes)
```

Comment:
IMO `final_size` is more accurate, because only `required_bytes` is what actually "required"

Reply:
done